records/generator.go

```diff
@@ -271,12 +271,15 @@ func (rg *RecordGenerator) InsertState(sj StateJSON, domain string, ns string,
 
 	// creates a map with slave IP addresses (IPv4)
 	rg.SlaveIPs = make(map[string]string)
+	rg.SRVs = make(rrs)
+	rg.As = make(rrs)
+
 	for _, slave := range sj.Slaves {
-		rg.SlaveIPs[slave.ID] = sanitizedSlaveAddress(slave.Hostname, spec)
+		ssa := sanitizedSlaveAddress(slave.Hostname, spec)
```

Comment:
actually there's a chance that `sanitizedSlaveAddress` doesn't return a string containing an ipv4 address, which will result in a bad A record. should probably test for that.